### PR TITLE
fix [#7368] Caffe converter test fails, causing CI to halt for all PRs 

### PR DESCRIPTION
**Pull request by joey2014**

### CODE REVIEW

File: tools/caffe_converter/convert_symbol.py [hunk 1]
```diff
@@ -128,6 +128,16 @@ def _parse_proto(prototxt_fname):
         param_string = ''
         skip_layer = False
         name = re.sub('[-/]', '_', layer.name)
+        for k in range(len(layer.bottom)):
```
Comment (piiswrong) [inline tools/caffe_converter/convert_symbol.py]: why move the code?

Comment (joey2014) [inline tools/caffe_converter/convert_symbol.py]: @piiswrong 
as comment on #7368 
The layer input name in some cases is changed which causes the layer added to group by mistake, for example, in vgg16, the layer fc6 input name pool5 is changed to flatten_0. Moving the state recording code to the begin of the function can fix the issue.

**Comment by piiswrong** [comment]
Thanks.
Could you revert this PR https://github.com/apache/incubator-mxnet/pull/7369/files to make sure it works?

**Comment by joey2014** [comment]
Yes, I have test it locally. The fix works 
INFO:root:test performance of model: vgg-16
INFO:root:./model/vgg-16_deploy.prototxt exists, skip to downloada
INFO:root:./model/vgg-16.caffemodel exists, skip to downloada
INFO:root:Saved checkpoint to "./model/vgg-16-0000.params"
INFO:root:speed : 145.695274 image/sec
INFO:root:('accuracy', 0.72265625)
INFO:root:('top_k_accuracy_5', 0.8984375)
INFO:root:test performance of model: resnet-50
INFO:root:./model/resnet-50_deploy.prototxt exists, skip to downloada
INFO:root:./model/resnet-50.caffemodel exists, skip to downloada
INFO:root:./model/resnet-50_mean.binaryproto exists, skip to downloada
INFO:root:Saved checkpoint to "./model/resnet-50-0000.params"
INFO:root:speed : 258.344333 image/sec
INFO:root:('accuracy', 0.75390625)
INFO:root:('top_k_accuracy_5', 0.92578125)

**Comment by szha** [comment]
Thanks, @joey2014. Please revert the removed test cases.

**Pull request merged**